File: android/feature/nova/src/main/kotlin/com/riskdetectedan/feature/nova/NovaPPEExampleScreen.kt
package com.riskdetectedan.feature.nova

import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import com.riskdetectedan.core.designsystem.isg.*

private const val PPE_EXAMPLE_FILE = "ISGADA_KKD_Zimmet_ve_Teslim_Formu_Duzenlenebilir.docx"

/** One bundled Word sample; no user details are sent to the PPE record service. */
@Composable
fun NovaPPEExampleScreen(onBack: () -> Unit) {
    val context = LocalContext.current
    LaunchedEffect(Unit) { com.riskdetectedan.core.data.nova.NovaForYouService.recordUse("ppe_form") }
    var message by remember { mutableStateOf<String?>(null) }
    val save = rememberLauncherForActivityResult(ActivityResultContracts.CreateDocument("application/vnd.openxmlformats-officedocument.wordprocessingml.document")) { uri ->
        if (uri != null) message = runCatching {
            context.assets.open("ppe_forms/$PPE_EXAMPLE_FILE").use { source ->
                context.contentResolver.openOutputStream(uri)?.use { target -> source.copyTo(target) }
                    ?: error("Unable to open destination")
            }
            "Word dosyası kaydedildi."
        }.getOrElse { "Word dosyası kaydedilemedi." }
    }
    val available = remember(context) { runCatching {
        context.assets.open("ppe_forms/$PPE_EXAMPLE_FILE").use { it.read() >= 0 }
    }.getOrDefault(false) }

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState())
        .padding(horizontal = 16.dp).padding(top = 12.dp, bottom = 24.dp + novaTabBarInset),
        verticalArrangement = Arrangement.spacedBy(14.dp)) {
        NovaListHeading("KKD Zimmet Formu Örneği", onBack)
        NovaHelpHint("Düzenlenebilir Word örneğini indirin, kendi işyerinizin bilgileriyle doldurun ve kullanın. Uygulama burada zimmet veya teslim kaydı oluşturmaz.")
        NovaCard(Modifier.fillMaxWidth(), padding = 16) {
            Column(verticalArrangement = Arrangement.spacedBy(9.dp)) {
                NovaText("KKD Zimmet ve Teslim Formu", style = NovaTypeToken.cardTitle)
                NovaText("Çalışan bilgileri, teslim edilen donanımlar, imzalar, ek teslim ve iade alanları içerir.", style = NovaTypeToken.meta)
                NovaText("Word · Düzenlenebilir örnek", style = NovaTypeToken.metaQuiet)
                if (available) NovaCompactActionButton("Word dosyasını indir", "arrow.down.doc", identifier = "ppe.example.download") {
                    message = null; save.launch(PPE_EXAMPLE_FILE)
                } else NovaText("Word dosyası bulunamadı", style = NovaTypeToken.metaQuiet)
            }
        }
        message?.let { NovaText(it, style = NovaTypeToken.meta) }
    }
}

File: App/DesignSystem/ISG/NovaPPEExampleScreen.swift
import SwiftUI

/// The KKD module offers one editable sample file and creates no handover record.
struct NovaPPEExampleScreen: View {
    let onBack: () -> Void

    private var documentURL: URL? {
        let name = "ISGADA_KKD_Zimmet_ve_Teslim_Formu_Duzenlenebilir"
        return Bundle.main.url(forResource: name, withExtension: "docx",
                               subdirectory: "PPEFormAssets/ppe_forms")
            ?? Bundle.main.url(forResource: name, withExtension: "docx")
    }

    var body: some View {
        content.onAppear { NovaForYouOutbox.recordUse("ppe_form") }
    }
    @ViewBuilder private var content: some View {
        NovaPageSurface(onEdgeBack: onBack) {
            ScrollView {
                VStack(alignment: .leading, spacing: 14) {
                    NovaPageHeading(title: RDLocalization.string("localizable.nova.ppeexample.screen.kkd.zimmet.formu.ornegi.8f8f826c", table: .localizable, fallback: "KKD Zimmet Formu Örneği"), onBack: onBack)
                    NovaHelpHint(text: RDLocalization.string("localizable.nova.ppeexample.screen.duzenlenebilir.word.ornegini.indirin.kendi.isyer.9360234c", table: .localizable, fallback: "Düzenlenebilir Word örneğini indirin, kendi işyerinizin bilgileriyle doldurun ve kullanın. Uygulama burada zimmet veya teslim kaydı oluşturmaz."))
                    NovaCard(padding: 16) {
                        VStack(alignment: .leading, spacing: 9) {
                            NovaText(text: RDLocalization.string("localizable.nova.ppeexample.screen.kkd.zimmet.ve.teslim.formu.2b5523f0", table: .localizable, fallback: "KKD Zimmet ve Teslim Formu"), style: .cardTitle)
                            NovaText(text: RDLocalization.string("localizable.nova.ppeexample.screen.calisan.bilgileri.teslim.edilen.donanimlar.imzal.be4d3f4d", table: .localizable, fallback: "Çalışan bilgileri, teslim edilen donanımlar, imzalar, ek teslim ve iade alanları içerir."), style: .meta)
                            NovaText(text: RDLocalization.string("localizable.nova.ppeexample.screen.word.duzenlenebilir.ornek.2b83a550", table: .localizable, fallback: "Word · Düzenlenebilir örnek"), style: .metaQuiet)
                            if let documentURL {
                                ShareLink(item: documentURL) {
                                    Label(RDLocalization.string("localizable.nova.ppeexample.screen.word.dosyasini.indir.019e3026", table: .localizable, fallback: "Word dosyasını indir"), systemImage: "square.and.arrow.up")
                                        .font(NovaFont.font(.bodyStrong))
                                }
                                .accessibilityIdentifier("ppe.example.download")
                            } else {
                                NovaText(text: RDLocalization.string("localizable.nova.ppeexample.screen.word.dosyasi.bulunamadi.0f2fbe98", table: .localizable, fallback: "Word dosyası bulunamadı"), style: .metaQuiet)
                            }
                        }
                        .frame(maxWidth: .infinity, alignment: .leading)
                    }
                }
                .padding(.horizontal, 16)
                .padding(.top, 4)
                .padding(.bottom, novaTabBarInset)
            }
        }
    }
}
